libplatsupport/src/plat/bcm2837/serial.c

```diff
@@ -78,5 +85,7 @@ int uart_init(const struct dev_defn* defn,
     dev->ioops      = *ops;
     dev->flags      = SERIAL_AUTO_CR;
 
+    *REG_PTR(dev->vaddr, MU_IER) |= MU_IER_ENABLE_READ_INTERRUPT;
```

Comment:
With this change, we are always enabling the interrupt. Are there any side effect with this that we have to consider for existing applications? Wouldn't it be better that is we expose a function here to enable or disable the interrupt, so application can control this? We could track the state via a dedicated bit in `dev->flags` if this is necessary.

Reply:
I reference other implementations in src/mach, those are all enable interrupt by default, I think may no other effect for this.

Reply:
Good point. Seems there is no specific pattern here, so we can also enable the interrupt by default. Might be more useful in the long run to have it than not to have it